bpython/curtsiesfrontend/repl.py

```diff
@@ -844,6 +846,13 @@ def on_control_d(self):
             self.current_line = (self.current_line[:self.cursor_offset] +
                                  self.current_line[(self.cursor_offset + 1):])
 
+    def on_control_o(self):
```

Comment:
Hm maybe we should fix the math instead, because in bash ctrl-o does work with the last line of history. You could change this by never doing `next_idx = -self.rl_history.index + 1`, and instead using the current index as described in the bottom comment - something like
~~~
def on_control_o(self):
    index = self.rl_history.index
    self.on_enter()  # this resets self.rlu_history.index
    self.rlu_history.index = index
    self._set_current_line(self.rlu_history.entries[-index))
~~~